contracts/Gate.sol

```diff
@@ -224,7 +224,7 @@ contract Gate is ProtocolVersioned, Owned {
     bytes32 _stakingIntentHash,
     bytes32 _unlockSecret)
     external
-    returns (bool /* success */)
+    returns (uint /* amount */)
```

Comment:
same comment as @deepesh-kn here, remove L265 and declare L227 as
`returns (uint256 stakeRequestAmount)`
note to prefer `uint256` over `uint`,
note that the suggested name `stakeRequestAmount`